src/components/dialog/dialogHelpers.ts

```diff
@@ -0,0 +1,97 @@
+import type { ButtonKind } from '../button/props'
+import DialogHelper from './dialogsContainer.svelte'
+import { writable } from 'svelte/store'
+
+let helper: DialogHelper | undefined
+export const ensureDialogHelper = () => {
+  if (helper) return
+
+  const mountPoint = document.createElement('div')
+  mountPoint.id = 'leo-dialog-helper'
+
+  document.body.appendChild(mountPoint)
+
+  helper = new DialogHelper({
+    target: mountPoint
+  })
+}
+
+interface Action {
+  result?: any
+  text: string
+  kind?: ButtonKind
+}
+
+interface AlertOptions {
+  title: string
+  body?: string
+  actions: Action[]
+}
+
+interface AlertInfo extends AlertOptions {
+  resolve: (result?: any) => void
+}
+
+export const dialogs = writable<AlertInfo[]>([])
+
+export const dialog = <T>(options: AlertOptions) => {
+  ensureDialogHelper()
+
+  let resolve: (result?: T) => void
+  const promise = new Promise((a) => (resolve = a))
+
+  const info = {
+    title: options.title,
+    body: options.body,
+    actions: options.actions,
+    resolve: (result?: T) => {
+      dialogs.update((d) => d.filter((i) => i !== info))
+      resolve(result)
+    }
+  }
+
+  dialogs.update((d) => [...d, info])
+
+  return promise
+}
+
+interface AlertDialogOptions {
+  okText?: string
+}
+
+export const alert = (message: string, options?: AlertDialogOptions) =>
+  dialog({
+    title: message,
+    actions: [
+      {
+        text: options?.okText ?? 'OK'
+      }
+    ]
+  })
+
+interface ConfirmDialogOptions {
+  okText?: string
+  cancelText?: string
+}
+
+export const confirm = (message: string, options?: ConfirmDialogOptions) =>
+  dialog({
+    title: message,
+    actions: [
+      {
+        result: false,
+        kind: 'outline',
+        text: options?.cancelText ?? 'Cancel'
```

Comment:
I believe(?) this is the first time we're introducing some default text in to a Leo component, so we should probably think about internationalisation. Perhaps a provider-based approach where consuming apps can provide the strings that all Leo components need, like:
```ts
type LeoStrings = { cancel: string, ok: string }

Leo.ProvideStrings = function(strings: LeoStrings)
```

Not sure if we can get away with sending the current locale only or send all locales and also tell leo which is the current locale...

Reply:
I agree, we should definitely think about this. For now though, I'm going to kick the can down the road and force you to provide strings.